lib/templates/lib/cloud_provider/gcp/google_translator_api.rb

```diff
@@ -0,0 +1,64 @@
+require "google/cloud/translate/v3"
+require 'dotenv/load'
+
+module CloudProvider
+  module Gcp
+
+    class GoogleTranslatorApi
+      @instance_mutex = Mutex.new
+      private_class_method :new
+
+      def self.instance
+        return @instance if @instance
+
+        @instance_mutex.synchronize do
+          @instance ||= new
+        end
+
+        @instance
+      end
+
+      def initialize
+        @client ||= ::Google::Cloud::Translate::V3::TranslationService::Client.new do |config|
+          config.credentials = ENV["KEYFILE_PATH"]
+        end
+      end
+
+      def translate(word, target_language_code)
+        begin
+          response = make_translate_text_request(word, target_language_code)
+
+          translated_word = response.translations.first&.translated_text || ""
+        rescue Exception => e
```

Comment:
What exceptions are expected here? Can specific ones only be handled? If API is not available at all, like with limit error we had seen earlier, it's better to stop requests than to continue I feel.
Thoughts?

Reply:
Yes I agree.